osu.Game/Scoring/ScoreInfoExtensions.cs

```diff
@@ -26,6 +27,35 @@ public static IEnumerable<ScoreInfo> OrderByTotalScore(this IEnumerable<ScoreInf
                      // Local scores may not have an online ID. Fall back to date in these cases.
                      .ThenBy(s => s.Date);
 
+        /// <summary>
+        /// Orders an array of <see cref="ScoreInfo"/>s by the selected <see cref="RankingsSort"/>.
+        /// </summary>
+        /// <param name="scores">The array of <see cref="ScoreInfo"/>s to reorder.</param>
+        /// <param name="rankingSort">The attribute to sort the scores by.</param>
+        /// <returns>The given <paramref name="scores"/> ordered by the selected mode.</returns>
+        public static IEnumerable<ScoreInfo> OrderByCriteria(this IEnumerable<ScoreInfo> scores, RankingsSort rankingSort)
+        {
+            switch (rankingSort)
+            {
+                case RankingsSort.Score:
+                    return scores.OrderByDescending(s => s.TotalScore);
+
+                case RankingsSort.Accuracy:
+                    return scores.OrderByDescending(s => s.Accuracy).ThenByDescending(s => s.TotalScore);
+
+                case RankingsSort.Combo:
+                    return scores.OrderByDescending(s => s.MaxCombo).ThenByDescending(s => s.TotalScore);
```

Comment:
This one is interesting because "sorting by combo", if it *is* extended to other places, might mean different things in different contexts. For instance, stable has a "combo" win condition in multiplayer, and as far as I can tell from source (haven't tested), it uses [current combo](https://github.com/peppy/osu-stable-reference/blob/fcda65f28f2c56263b3fd2b411579480b0216efa/osu!/GameModes/Play/Components/ScoreboardEntry.cs#L161) rather than max combo for sorting.

Maybe not particularly relevant for this PR, but may be relevant for deciding on how far to extend this feature.